dataproc_jupyter_plugin/handlers.py

```diff
@@ -152,6 +154,15 @@ async def post(self):
         input_data = self.get_json_body()
         project_id = input_data["projectId"]
         region = input_data["region"]
+        # Validate inputs before processing
+        if not re.fullmatch(constants.PROJECT_REGEXP, project_id):
+            self.set_status(400)
+            self.finish({"error": f"Invalid project ID: {project_id}"})
```

Comment:
Please change this to "Unsupported" instead of "Invalid".

That way if the naming rules ever change in the future, or if we find a valid value that doesn't match our regex, the error message is more correct.